test/basic.html

```diff
@@ -206,6 +206,211 @@
+              tabs.autoselectDelay = -1;
+              tabs.select(0);
+              MockInteractions.keyDownOn(tabs.selectedItem, RIGHT);
+              assert.equal(tabs.selected, 0);
+              Polymer.Base.async(function() {
+                assert.equal(tabs.selected, 1);
+
+                MockInteractions.keyDownOn(tabs.selectedItem, RIGHT);
+                assert.equal(tabs.selected, 1);
+                Polymer.Base.async(function() {
+                  assert.equal(tabs.selected, 2);
+
+                  MockInteractions.keyDownOn(tabs.selectedItem, LEFT);
+                  assert.equal(tabs.selected, 2);
+                  Polymer.Base.async(function() {
+                    assert.equal(tabs.selected, 1);
+                    done();
+                  });
+                });
+              });
+            });
+          });
+
+        test('with autoselect-delay zero, tabs are selected with ' +
+          'microtask timing after the keyup', function(done) {
+            ensureDocumentHasFocus();
+            Polymer.Base.async(function() {
+              tabs.autoselectDelay = 0;
+              tabs.select(0);
+
+              MockInteractions.keyDownOn(tabs.selectedItem, RIGHT);
+              Polymer.Base.async(function() {
+                assert.equal(tabs.selected, 0);
+                assert.equal(tabs.items.indexOf(tabs.focusedItem), 1);
+
+                // No keyup between keydown events: the key is being held.
+                MockInteractions.keyDownOn(tabs.selectedItem, RIGHT);
+                Polymer.Base.async(function() {
+                  assert.equal(tabs.selected, 0);
+                  assert.equal(tabs.items.indexOf(tabs.focusedItem), 2);
+
+                  MockInteractions.keyUpOn(tabs.selectedItem, RIGHT);
+                  assert.equal(tabs.selected, 0);
+                  Polymer.Base.async(function() {
+                    assert.equal(tabs.selected, 2);
+                    assert.equal(tabs.items.indexOf(tabs.focusedItem), 2);
+
+                    MockInteractions.keyDownOn(tabs.selectedItem, LEFT);
+                    Polymer.Base.async(function() {
+                      assert.equal(tabs.selected, 2);
+                      assert.equal(tabs.items.indexOf(tabs.focusedItem), 1);
+
+                      MockInteractions.keyUpOn(tabs.selectedItem, LEFT);
+                      assert.equal(tabs.selected, 2);
+                      Polymer.Base.async(function() {
+                        assert.equal(tabs.selected, 1);
+                        done();
+                      });
+                    });
+                  });
+                });
+              });
+            });
+          });
+
+        test('with autoselect-delay positive, tabs are selected with ' +
+          'microtask timing after the keyup and delay', function(done) {
+            ensureDocumentHasFocus();
+            Polymer.Base.async(function() {
+              var DELAY = 500;
```

Comment:
Can we make this delay v small (like `1`) so that the test doesn't run this long?